### ISSUE TITLE
construction of generic class conflicts with "less than" operator in if condition

### ISSUE THREAD

**Issue by felixhao28** [atom/language-java#184]
### Description

![image](https://user-images.githubusercontent.com/2417391/50687111-fe3ca880-105a-11e9-95ea-650ddd5a9edf.png)

```java
class A {
    void f() {
        int X = 5;
        if (X < 100)
            isTokenPerNode = false;
        for (int A : list) {
            // ...
        }
        if (new HashSet<String>().size() < 100)
            isTokenPerNode = false;
        for (int A : list) {
            // ...
        }
    }
}
```

If the condition contains a generic type, the comparator "<" won't be parsed correctly. And it also breaks the rest of the file.

Version: 
7c3d0d3e37920790cd3afb4e68830bdb215851a3

**Comment by sadikovi** [comment]
Relates to #132

**Comment by felixhao28** [comment]
OK. Feel free to close it if it's duplicate.

**Comment by Vigilans** [comment]
This issue has been solved in #214:
![image](https://user-images.githubusercontent.com/20227484/69945653-00d2a700-1525-11ea-9e39-a04a5a67272c.png)

It is actually a variant of #213: The dot punctuation '.' didn't stop the new expression. 

Effect of removing the dot operator from end regex:
![image](https://user-images.githubusercontent.com/20227484/69945835-5dce5d00-1525-11ea-9fb6-8ab273bfeee9.png)

**Comment by sadikovi** [comment]
I double checked that this issue has been fixed in master.
